utils.js

```diff
@@ -227,5 +254,65 @@ module.exports = {
     // The legacy method 'expiration()' will set the policy TTL value via the Cache-Control max-age value,
     // so no conflicts here.
     props.expiration = policy.storable() ? Math.round(policy.timeToLive() / 1000) : 0;
+  },
+
+  /**
+   * Returns the `expiration` (TTL) value calculated as a minimum of the `props.expiration` and `policy.timeToLive`.
+   * The resulting value is multiplied by `2`due to enable further handling of the stale cache entries,
+   * (when the policy allows for that).
+   * The resulting value is to be used for underlying cache implementation.
+   *
+   * @param {object} props  - The request-basis properties, which affect cache behavior.
+   * @param {object} policy - The cache policy.
+   * @returns {number} The expiration (TTL) time in seconds.
+   */
+  getExpiration: function (props, policy) {
+    return props.expiration
```

Comment:
this could be simplified to `Math.min(props.expiration * 2 || Number.MAX_VALUE, Math.round(policy.timeToLive() * 2 / 1000))` Personal preference though